Detectors/FIT/FT0/simulation/src/Digits2Raw.cxx

```diff
@@ -0,0 +1,124 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "Headers/RAWDataHeader.h"
+#include "CommonDataFormat/InteractionRecord.h"
+#include "DataFormatsFT0/RawEventData.h"
+#include "FT0Simulation/Digits2Raw.h"
+#include <TStopwatch.h>
+#include <cassert>
+#include <fstream>
+#include <vector>
+#include <bitset>
+#include <iomanip>
+#include "TFile.h"
+#include "TTree.h"
+
+using namespace o2::ft0;
+
+ClassImp(Digits2Raw);
+//Digits2Raw(char * fileRaw, std::string fileDigitsName)
+void Digits2Raw::readDigits(const char* fileRaw, const char* fileDigitsName)
+{
+  std::cout << "**********Digits2Raw::convertDigits" << std::endl;
+  std::vector<o2::ft0::Topo> lut_data(NCHANNELS_PM * NPMs);
+  for (int link = 0; link < NPMs; ++link)
+    for (int quadrant = 0; quadrant < NCHANNELS_PM; ++quadrant)
+      lut_data[link * NCHANNELS_PM + quadrant] = o2::ft0::Topo{link, quadrant};
+  o2::ft0::LookUpTable lut{lut_data};
+
+  std::cout << " ##### LookUp set " << std::endl;
+  mFileDest.exceptions(std::ios_base::failbit | std::ios_base::badbit);
+  mFileDest.open(fileRaw);
+  // mFileDest = fopen(fileRaw, "wb+");
+
+  TFile* fdig = TFile::Open(fileDigitsName);
+  assert(fdig != nullptr);
+  std::cout << " Open digits file " << std::endl;
+  TTree* digTree = (TTree*)fdig->Get("o2sim");
+  std::vector<o2::ft0::Digit>* digArr = new std::vector<o2::ft0::Digit>;
+  digTree->SetBranchAddress("FT0Digit", &digArr);
+  Int_t nevD = digTree->GetEntries(); // digits in cont. readout may be grouped as few events per entry
+  std::cout << "Found " << nevD << " events with digits " << std::endl;
+  for (Int_t iev = 0; iev < nevD; iev++) {
+    digTree->GetEvent(iev);
+    for (const auto& digit : *digArr) {
+      convertDigits(digit, lut);
+    }
+  }
+}
+
+/*******************************************************************************************************************/
+void Digits2Raw::convertDigits(const o2::ft0::Digit& digit, const o2::ft0::LookUpTable& lut)
+{
+  auto mIntRecord = digit.getInteractionRecord();
+  std::vector<o2::ft0::ChannelData> mTimeAmp = digit.getChDgData();
+  bool is0TVX = digit.getisVrtx();
+  int oldlink = -1;
+  int nlink = 0;
+  int nchannels = 0;
+  for (auto& d : mTimeAmp) {
+    nlink = lut.getLink(d.ChId);
+    if (nlink != oldlink) {
+      flushEvent(oldlink, mIntRecord, nchannels);
+      oldlink = nlink;
+      nchannels = 0;
+    }
+    mEventData[nchannels].channelID = lut.getQuadrant(d.ChId);
+    mEventData[nchannels].charge = 2.2857143 * d.QTCAmpl; //7 mV ->16channels
```

Comment:
It is better to avoid such a hardcoded numbers, can you define them as constants of the class (if used elsewhere, put in some common header)